src/js/pages/auth/Callback.jsx

```diff
@@ -20,31 +19,53 @@ export default () => {
     };
 
     getTokenSilently()
-      .then(token => {
+      .then(async (token) => {
         const api = buildApi(token);
         const inviteLink = query.get('inviteLink');
 
         if (inviteLink) {
           const body = new InvitationLink(inviteLink);
-          api.checkInvitation(body)
-            .then(check => {
-              if (check.valid && check.active) {
-                const invType = check.type;
-                return api.acceptInvitation(body);
-              }
-
-              return Promise.reject(new Error('invalid invitation'));
-            })
-            .then(resp => setRedirectTo(query.get('targetUrl')))
-            .catch(err => setErrorMessage(err.message));
-        } else {
-          setRedirectTo(query.get('targetUrl'));
+
+          try {
+
+            let check;
+            try {
+              check = await api.checkInvitation(body);
+            } catch (err) {
+              throw new Error(`${err.error.message}`);
+            }
+
+            if (!check.valid || !check.active) {
+              const cause = !check.valid ? 'valid' : 'active';
+              throw new Error(`Invitation is not ${cause}`);
+            }
+
+            let acceptance;
+            try {
+              const acceptance = await api.acceptInvitation(body);
+            } catch (err) {
+              throw new Error(`Could not accept the invitation. Err#${err.body.status} ${err.body.type}. ${err.body.detail}`);
+            }
+
+          } catch (err) {
+            setErrorMessage(err.message);
+
+            // TODO (dpordomingo):
+            // This is needed because if the invitation fails, the user should not be logged in.
+            // Also, the logout is delayed because Auth0 redirects to the 'logoutRedirectUri'
+            // inmediately, and then the user would not be able to see the error message.
+            // Workaround: store the error, let Auth0 to redirect, and then show the stored message.
+            window.setTimeout(logout, 3000);
```

Comment:
logout({ returnTo: window.ENV.auth.logoutRedirectUri });
```suggestion
            window.setTimeout(function() {
                logout({ returnTo: window.ENV.auth.logoutRedirectUri });
            }, 3000);
```

Reply:
why? Since the `Allowed Logout URLs` must be configured at application level, we can rely on that value as the default redirect after the logout (from the [Auth0 logout docs](https://auth0.com/docs/api/authentication?javascript#logout)), can't we?

I also saw it's also suggested that way (without the `returnTo` optional argument) in the [Auth0 React guide](https://auth0.com/docs/quickstart/spa/react#create-the-navbar-component)

Reply:
In theory, there could be more than one `Allowed Logout URLs`, so I think it would be better to specify it explicitly.

Reply:
Yes, `Allowed Logout URLs` can contain many different logout urls, and it takes as default the first one according to the docs.

I feel that it would be better to let the default behave like a real default, and explicitly override it when it's really needed.
But if you really prefer to logout with the explicit `redirectTo` just confirm it, and I'll commit.

Reply:
Yup, so that in case we change for example the configuration we don't need to change the webapp.  It would remove the dependency 👍

Reply:
That's most probably a good counter-example. If we change the config we'll most probably need to change the web app because the old `redirectTo` would be no longer accepted by the new `Allowed Logout URLs` config.
That would not happen if we would rely on the default behavior.

But since it's an easy change, let's merge this following your preference, and review this in the future if we need it.